src/Services/Payment/Payment.php

```diff
@@ -119,18 +119,19 @@ public function requestPayment($amount, $reason = 'Purchase') : string
             // Payment request was successful, continue redirect
             $responseData = $response->getData();
             header("Location: " . $responseData['url']);
+            return self::RETURN_SUCCESS;
 
         } elseif ($response->isRedirect()) {
             // Redirect to offsite payment gateway
             $response->redirect();
+            return ''; // TODO: should there be a return value for "redirecting"?
```

Comment:
Perhaps a warning, since the only way to see this would be if the redirect failed, which could indicate an interruption in the process. Maybe `const RETURN_REDIRECT = 'warning4';` with a message of `__('Your online payment redirect was interrupted before it could be completed. Please try again and if the error persists, contact the school.')`

```suggestion
            return 'self::RETURN_REDIRECT';
```

Reply:
I added the message accordingly.

But I have doubt. From my brief read to Omnipay documentation, the redirect might not be an error, but could be an intermediate state. Some payment gateways (e.g. PayPal) requires merchant site to redirect payers to them when doing payment. So the message here might not match what is actually happening.